src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/assets_transfers/per_recipient/per_recipient_asset_transfers.rs

```diff
@@ -0,0 +1,176 @@
+        Self::new(
+            AccountAddress::sample_mainnet(),
+            [
+                PerRecipientAssetTransfer::sample_mainnet(),
+                PerRecipientAssetTransfer::sample_mainnet_other(),
+            ],
+        )
+    }
+
+    pub(crate) fn sample_mainnet_other() -> Self {
+        Self::new(
+            AccountAddress::sample_mainnet_other(),
+            [PerRecipientAssetTransfer::sample_mainnet_other()],
+        )
+    }
+
+    pub(crate) fn sample_stokenet() -> Self {
+        Self::new(
+            AccountAddress::sample_stokenet(),
+            [
+                PerRecipientAssetTransfer::sample_stokenet(),
+                PerRecipientAssetTransfer::sample_stokenet_other(),
+            ],
+        )
+    }
+
+    pub(crate) fn sample_stokenet_other() -> Self {
+        Self::new(
+            AccountAddress::sample_stokenet_other(),
+            [PerRecipientAssetTransfer::sample_stokenet_other()],
+        )
+    }
+}
+
+impl HasSampleValues for PerRecipientAssetTransfers {
+    fn sample() -> Self {
+        Self::sample_mainnet()
+    }
+
+    fn sample_other() -> Self {
+        Self::sample_stokenet_other()
+    }
+}
+
+impl PerRecipientAssetTransfers {
+    /// Transpose: `PerRecipient` -> `PerAsset`
+    pub fn transpose(&self) -> PerAssetTransfers {
+        let mut per_asset_fungibles = IndexMap::<
+            ResourceAddress,
+            PerAssetTransfersOfFungibleResource,
+        >::new();
+
+        let mut per_asset_non_fungibles = IndexMap::<
+            ResourceAddress,
+            PerAssetTransfersOfNonFungibleResource,
+        >::new();
+
+        self.transfers.clone().into_iter().for_each(|t| {
+            let x = t.clone();
+            let recipient = &x.recipient;
+
+            x.fungibles.clone().into_iter().for_each(|y| {
+                if let Some(existing_transfers) =
+                    per_asset_fungibles.get_mut(&y.resource_address)
+                {
+                    existing_transfers.expanded((recipient, y.clone()));
+                } else {
+                    per_asset_fungibles.insert(
+                        y.resource_address.clone(),
+                        PerAssetTransfersOfFungibleResource::new(
+                            PerAssetFungibleResource::new(
+                                y.clone().resource_address,
+                                y.clone().divisibility,
+                            ),
+                            [PerAssetFungibleTransfer::from((
+                                recipient,
+                                y.clone(),
+                            ))],
+                        ),
+                    );
+                }
+            });
+
+            x.non_fungibles.clone().into_iter().for_each(|y| {
+                if let Some(existing_transfers) =
+                    per_asset_non_fungibles.get_mut(&y.resource_address)
+                {
+                    existing_transfers.expanded((recipient, y.clone()));
```

Comment:
this is the second transfer of that resource, expand the existing list of transfers